sympy/physics/vector/frame.py

```diff
@@ -1558,8 +1558,10 @@ def partial_velocity(self, frame, *gen_speeds):
 
         """
 
-        partials = [self.ang_vel_in(frame).diff(speed, frame, var_in_dcm=False)
-                    for speed in gen_speeds]
+        from sympy.physics.vector.functions import partial_velocity
```

Comment:
Are these imports here because of some circular dependency issue? If not, then they should be at the top of the file with the other imports. If it is a circular dependency then we need to rethink where the primary function should be stored.

Reply:
> Are these imports here because of some circular dependency issue? If not, then they should be at the top of the file with the other imports. If it is a circular dependency then we need to rethink where the primary function should be stored.

Yes, I put them inside because I was getting a circular dependency issue, but only for the frame class. However for consistency I put the imports inside the methods definition for both classes. What's the procedure in these cases?

Reply:
We can keep the imports like this if needed, but identifying why it occurs and fixing it can be better. I guess the partial_velocity function depends on things in frame.

Reply:
Procedure is to only put imports within functions/methods if you have to because you can't fix the circular dependency or maybe if you are trying to somehow lazy load a module for specific import performance reasons.

Reply:
I guess this is because functions.py imports things from frame and not point.

Reply:
Maybe the circular dependency issue is the reason why the code was "duplicated" in both classes and partial_velocity() from functions.py was not used

Reply:
Possibly!

Reply:
You can also go back to that. We could just open an issue about the circular dependencies.

Reply:
> You can also go back to that. We could just open an issue about the circular dependencies.

Opened #26411

A possible fix for these circular imports would be to create a `Vector.partial_diff(self, vars, linear=False)` method.